src/services/httpService.ts

```diff
@@ -0,0 +1,72 @@
+import axios, {
+  type AxiosInstance,
+  type AxiosRequestConfig,
+  type AxiosResponse,
+  type InternalAxiosRequestConfig,
+} from 'axios'
+import Cookies from 'js-cookie'
+
+interface AdminInfo {
+  token: string
+}
+
+const baseURL: string = `${import.meta.env.VITE_APP_API_BASE_URL}`
+const instance: AxiosInstance = axios.create({
+  baseURL,
+  timeout: 50000,
+  headers: {
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+  },
+})
+
+instance.interceptors.request.use(function (config: InternalAxiosRequestConfig) {
+  let adminInfo: AdminInfo | null = null
+  const adminInfoCookie = Cookies.get('adminInfo')
+
+  if (adminInfoCookie) {
+    try {
+      adminInfo = JSON.parse(adminInfoCookie) as AdminInfo
+    } catch (error) {
+      console.error('Error parsing adminInfo cookie:', error)
+    }
+  }
+
+  let company: string | undefined = Cookies.get('company')
+  if (adminInfo) {
+    config.headers['authorization'] = `Bearer ${adminInfo.token}`
+  }
+  if (company) {
+    config.headers['company'] = company
+  }
+  return config
+})
+
+const responseBody = <T>(response: AxiosResponse<T>): T => response.data
+
+interface Requests {

```

Comment:
just one small observation here, let's move this interface to the top of the file a little higher (where the other interface is declared.